### ISSUE TITLE
Newsletter categories: Create POST (or PUT?) endpoint to subscribe/unsubscibe from categories

### ISSUE THREAD

**Issue by TimBroddin** [Automattic/wp-calypso#80046]
Create an endpoint that takes one or more term_ids and a boolean for each of them that indicates if the user wants to subscribe or unsubscribe from them.

In the endpoint, loop through the array:
- if subscribe = `true` => remove any exclusion that might be present
- if subscribe = `false` => add an exclusion for this category

Example payload:

```
[ { term_id: 13, subscribe: true }, { term_id: 14, subscribe: false } ]
```

**Comment by TimBroddin** [comment]
Draft PR: D118015-code

**Comment by TimBroddin** [comment]
Landed in this one: D117823-code
